src/inc/til/rect.h

```diff
@@ -841,6 +680,32 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
         {
             return { left, top, right, bottom };
         }
+
+        // til::rect and RECT have the exact same layout at the time of writing,
+        // so this function lets you unsafely "view" this rect as a RECT
+        // if you need to pass it to a Win32 function.
+        //
+        // Use as_win32_rect() as sparingly as possible because it'll be a pain to hack
+        // it out of this code base once til::rect and RECT aren't the same anymore.
+        // Prefer casting to RECT and back to til::rect instead if possible.
+        RECT* as_win32_rect() noexcept
+        {
+#pragma warning(suppress : 26490) // Don't use reinterpret_cast (type.1).
+            return std::launder(reinterpret_cast<RECT*>(this));
+        }
+
+        // til::rect and POINT[2] have the exact same layout at the time of writing,

```

Comment:
now *this* is horrifying.

can you static_assert that they have the same layout somehow

Reply:
Yeah I have!